book/visual-effects.md

```diff
@@ -1345,97 +1329,81 @@ def paint_visual_effects(node, cmds, rect):
     ]
 ```
 
-[^except-filters]: This works for opacity, but not for filters that
-"move pixels" such as [blur]. Such a filter needs to be applied before clipping,
-not when blending into the parent surface. Otherwise, the edge of the blur will
-not be sharp.
-
-[blur]: https://developer.mozilla.org/en-US/docs/Web/CSS/filter-function/blur()
-
-There's one more important optimization to make: getting rid of the
-destination-in compositing surface for rounded corners. While our current
-approach works just fine, and is a good idea for general masks, rounded corners
-are so common on the web that Skia has a special `clipRRect` command just for
-this use case.
-
-There are multiple advantages to using `clipRRect` over an explicit
-destination-in surface. First, it allows Skia to internally optimize away even
-more surfaces in common situations, replacing them with an equivalent
-implementation directly on the GPU, via a shader.[^shader-rounded] Second,
-`clipRRect` makes it much easier for Skia to skip subsequent draw operations
-that don't intersect the rounded rectangle,[^see-chap-1] or dynamically draw
-only the parts of commands that intersect it.
-
-[^shader-rounded]: GPU programs are out of scope for this book, but if you're
-curious there are many online resources describing ways to to do this. Skia
-of course also has an implementation in its GPU-accelerated code paths.
-
-[^see-chap-1]: This is basically the same optimization we added in Chapter
-2 to avoid painting off-screen text.
-
-To use `clipRect`, we'll also need another method: `save`. Once you call a
-method like `clipRect`, all subsequent canvas commands are clipped, until you
-tell Skia to stop. The way to make it stop is with `save` and `restore`---you
-call `save` before calling `clipRect`, and `restore` after finishing drawing
-the commands that should be clipped. `save` means "snapshot the current
-state of the canvas", and `restore` rolls back to the most recent snapshot.
-Here's an example:
+There's one more optimization that can make clipping less expensive.
+Right now, we might draw a very large and complex shape, and then clip
+all of it out.[^not-really-here] That means Skia wastes time drawing
+pixels it doesn't need. To avoid this, Skia has a special `clipRRect`
+command just for this use case. This `clipRRect` command takes in a
+rounded rectangle, changes the *canvas state* so that all future
+commands skip drawing any pixels outside it.
+
+[^not-really-here]: Given our browser's limited layout capabilities,
+    it's kind of hard to come up with an example of this, but complex
+    diacritics could have this effect. In a real browser, where
+    clipping applies in more cases, this optimization is even more
+    important.
+
+When you use `clipRRect`, Skia dynamically draws only the parts of a
+shape that intersection the given rounded rectangle, via specialized
+code in its shape implementations.[^shader-rounded] Skia can also to
```

Comment:
Your new text repeats itself about avoiding clipped-out stuff. Not clear to me if the new phrasing is a big improvement on the old text..